python/ray/autoscaler/v2/instance_manager/instance_manager.py

```diff
@@ -63,3 +66,215 @@ def get_instance_manager_state(
             The reply to the request.
         """
         pass
+
+
+class InstanceUtil:
+    """
+    A helper class to group updates and operations on an Instance object defined
+    in instance_manager.proto
+    """
+
+    @staticmethod
+    def new_instance(
+        instance_id: str,
+        instance_type: str,
+        resources: Dict[str, float],
+        request_id: str = "",
+    ) -> Instance:
+        instance = Instance()
+        instance.version = 0  # it will be populated by the underlying storage.
+        instance.instance_id = instance_id
+        instance.instance_type = instance_type
+        instance.launch_request_id = request_id
+        instance.total_resources.update(resources)
+        instance.status = Instance.QUEUED
+        InstanceUtil._record_status_transition(
+            instance, Instance.QUEUED, "created from InstanceUtil"
+        )
+        return instance
+
+    @staticmethod
+    def is_cloud_instance_allocated(instance: Instance) -> bool:
+        """
+        Returns True if the instance is allocated by the cloud provider, i.e.
+        there's already a cloud node for the instance.
+        """
+        assert instance.status != Instance.UNKNOWN
+        return instance.status in {
+            Instance.ALLOCATED,
+            Instance.RAY_INSTALLING,
+            Instance.RAY_RUNNING,
+            Instance.RAY_STOPPING,
+            Instance.RAY_STOPPED,
+            Instance.STOPPING,
+            Instance.RAY_INSTALL_FAILED,
+        }
+
+    @staticmethod
+    def is_ray_pending(instance: Instance) -> bool:
+        """
+        Returns True if the instance is pending ray installation or ray startup,
+        regardless of whether the instance is allocated by the cloud provider.
```

Comment:
Change the comment saying that it should include all states that can be transitioned into RAY_RUNNING.

Rename the function as well to reflect that.